src/Microsoft.Framework.Runtime/Servicing/Breadcrumbs.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Reflection;
+using NuGet;
+
+namespace Microsoft.Framework.Runtime.Servicing
+{
+    public class Breadcrumbs
+    {
+        private readonly string _breadcrumbsFolder;
+
+        public Breadcrumbs()
+            : this(ResolveBreadcrumbsFolder())
+        {
+        }
+
+        public Breadcrumbs(string breadcrumbsFolder)
+        {
+            // If the directory doesn't exist, don't create it because it
+            // needs special permissions on it
+            _breadcrumbsFolder = Directory.Exists(breadcrumbsFolder) ? breadcrumbsFolder : null;
```

Comment:
It's kind of a matter of style, but this could be a simple `if()` statement. If the directory doesn't exist no need to set the folder to null because it's already null. Might be easier to debug.